### PR TITLE
SQSERVICES-1169 New internal endpoint to configure the guest links team feature

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/SQSERVICES-1169

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If HTTP endpoint paths have been added or renamed, the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/galley/src/Galley/API/Teams/Features.hs [hunk 1]
```diff
@@ -632,18 +633,19 @@ getGuestLinkInternal ::
...
-      (Public.Unlocked, Just featureStatus) ->
+    pure $ case maybeFeatureStatus of
+      Just featureStatus ->
         Public.TeamFeatureStatusNoConfigAndLockStatus
           (Public.tfwoStatus featureStatus)
-          Public.Unlocked
-      (Public.Unlocked, Nothing) -> cfgDefault {Public.tfwoapsLockStatus = Public.Unlocked}
-      (Public.Locked, _) -> cfgDefault {Public.tfwoapsLockStatus = Public.Locked}
+          defLockStatus
+      Nothing -> cfgDefault
```
Comment (fisx) [inline services/galley/src/Galley/API/Teams/Features.hs:648]: I think this is a genuine fix, but it's still not completely correct.  Shouldn't the lock status in L647 come from cassandra, not the default?

(Wow, this is complicated, though.  Why is this so complicated?)

Comment (fisx) [reply services/galley/src/Galley/API/Teams/Features.hs:648]: (And have we been making the same mistake elsewhere?)

**Review by fisx — APPROVED** [review]

File: services/galley/schema/src/Main.hs [hunk 2]
```diff
@@ -103,7 +104,8 @@ main = do
       V53_AddRemoteConvStatus.migration,
       V54_TeamFeatureSelfDeletingMessages.migration,
       V55_SelfDeletingMessagesLockStatus.migration,
-      V56_GuestLinksTeamFeatureStatus.migration
+      V56_GuestLinksTeamFeatureStatus.migration,
+      V57_GuestLinksLockStatus.migration
       -- When adding migrations here, don't forget to update
       -- 'schemaVersion' in Galley.Cassandra
```
Comment (fisx) [inline services/galley/schema/src/Main.hs:110]:
```
       -- When adding migrations here, don't forget to update
       -- 'schemaVersion' in Galley.Cassandra
```

File: services/galley/schema/src/V57_GuestLinksLockStatus.hs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import Cassandra.Schema
+import Imports
+import Text.RawString.QQ
+
+migration :: Migration
+migration = Migration 57 "Add lock status for guest links team feature" $ do
+  schema'
+    [r| ALTER TABLE team_features ADD (
+          guest_links_lock_status int
```
Comment (fisx) [inline services/galley/schema/src/V57_GuestLinksLockStatus.hs:31]: should this be consistent with the symbol?

```
type KnownTeamFeatureNameSymbol 'TeamFeatureGuestLinks = "conversationGuestLinks"
```

File: services/galley/src/Galley/API/Teams/Features.hs [hunk 4]
```diff
@@ -635,15 +634,14 @@ getGuestLinkInternal = \case
...
+    pure $ case (lockStatus, mbFeatureStatus) of
       (Public.Unlocked, Just featureStatus) ->
         Public.TeamFeatureStatusNoConfigAndLockStatus
           (Public.tfwoStatus featureStatus)
-          Public.Unlocked
-      (Public.Unlocked, Nothing) -> cfgDefault {Public.tfwoapsLockStatus = Public.Unlocked}
-      (Public.Locked, _) -> cfgDefault {Public.tfwoapsLockStatus = Public.Locked}
+          lockStatus
+      (Public.Unlocked, Nothing) -> cfgDefault {Public.tfwoapsLockStatus = lockStatus}
+      (Public.Locked, _) -> cfgDefault {Public.tfwoapsLockStatus = lockStatus}
```
Comment (fisx) [inline services/galley/src/Galley/API/Teams/Features.hs:644]: [nit-pick] not sure i like this change.  it's less direct.  (then again it's more resilient to change in the pattern, possibly.)

**Review by fisx — CHANGES_REQUESTED** [review]
close!  :)

File: services/galley/schema/src/Main.hs (hunk 2 above)
Comment (battermann) [reply services/galley/schema/src/Main.hs:110]: done

**Review by fisx — APPROVED** [review]

**Pull request merged**